### PR TITLE
Fix compatibility with latest version of setuptools

### PR DESCRIPTION
**Pull request by astrofrog**
This gets the build running but there are real errors, so I will investigate further.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,29 +23,15 @@ env:
     - PYTHON_VERSION=3.4
     - PYTHON_VERSION=3.5
     - PYTHON_VERSION=3.6 EVENT_TYPE='push pull_request cron'
-
+    - PYTHON_VERSION=2.7 SETUPTOOLS_VERSION=dev DEBUG=True
+      CONDA_DEPENDENCIES='pytest sphinx cython numpy packaging six appdirs'
```
Comment (bsipocz) [inline .travis.yml]: pytest and pip are default installs by ci-helpers, no need to list them again

**Comment by bsipocz** [comment]
@astrofrog - Is this ready to be merged?

File: .travis.yml (hunk 1 above)
Comment (astrofrog) [inline .travis.yml]: Fixed

**Comment by astrofrog** [comment]
@bsipocz - yes, once tests pass (had to rebase)

**Comment by bsipocz** [comment]
I'll try to backport this to 1.3.2, though doubt that it would solve any of the issues...

**Comment by bsipocz** [comment]
@astrofrog - Is it necessary to test everything on both osx and linux?

**Comment by astrofrog** [comment]
@bsipocz - I've seen some setuptools issues with just MacOS X, so I was just trying to be on the safe side, but we could potentially reduce the number of builds per platforms (e.g. the SETUPTOOLS_VERSION=20 build could maybe be combined with one of the other ones)

**Comment by bsipocz** [comment]
OK, that sounds good. I'll also move the sphinx 1.6+ build to one platform as that one is only a temporary one.

**Pull request merged**